src/deploy/durable_objects.rs

```diff
@@ -0,0 +1,337 @@
+                &DurableObjectCreateNSRequest {
+                    name: ns.namespace_name.clone(),
+                    implementation: Some(DurableObjectNSImpl {
+                        script: self.script_name.clone(),
+                        class: ns.class_name.clone(),
+                    }),
+                },
+                &self.account_id,
+                user,
+            )?;
+        }
+
+        for (id, ns) in update_namespaces {
+            updated_namespaces.push(ns.namespace_name.clone());
+            update_ns(
+                &id,
+                &DurableObjectNSImpl {
+                    script: self.script_name.clone(),
+                    class: ns.class_name.clone(),
+                },
+                &self.account_id,
+                user,
+            )?;
+        }
+
+        Ok(updated_namespaces)
+    }
+
+    fn get_existing_namespaces(&self, user: &GlobalUser) -> Result<(), failure::Error> {
+        log::info!("getting existing durable objects");
+        let client = http::legacy_auth_client(user);
+        self.existing_namespaces
+            .replace(list_namespaces_by_name_to_id(&client, &self.account_id)?);
+        Ok(())
+    }
+
+    fn init_self_referential_namespaces(&self, user: &GlobalUser) -> Result<(), failure::Error> {
+        log::info!("initializing self-referential namespaces");
+        let implemented_namespace_names = self
+            .durable_objects
+            .uses
+            .iter()
+            .flat_map(|nses| nses)
+            .filter_map(|ns| ns.namespace_name.as_ref())
+            .collect::<Vec<_>>();
+
+        let existing_namespaces = self.existing_namespaces.borrow();
+        let existing_namespace_names = existing_namespaces.keys().collect::<Vec<_>>();
+
+        let new_self_referential_namespace_names = self
+            .durable_objects
+            .uses
+            .iter()
+            .flat_map(|nses| nses)
+            .filter_map(|ns| {
+                ns.namespace_name.as_ref().and_then(|name| {
+                    if implemented_namespace_names.contains(&name)
+                        && !existing_namespace_names.contains(&name)
+                    {
+                        Some(name)
+                    } else {
+                        None
+                    }
+                })
+            });
+
+        for name in new_self_referential_namespace_names {
+            log::info!("creating error namespace {}", name);
+            let new_ns = create_ns(
+                &DurableObjectCreateNSRequest {
+                    name: name.clone(),
+                    implementation: None,
+                },
+                &self.account_id,
+                user,
+            )?;
+            self.existing_namespaces
+                .borrow_mut()
+                .insert(new_ns.id.clone(), new_ns);
+        }
+
+        Ok(())
+    }
+
+    fn hydrate_target_with_ns_ids(&self, target: &mut Target) -> Result<(), failure::Error> {
```

Comment:
if we could provide namespace names instead of ids to the api, we wouldn't need to do this.